roscala/src/main/scala/coop/rchain/roscala/ob/TblObject.scala

```diff
@@ -1,12 +1,23 @@
 package coop.rchain.roscala.ob
 
-import scala.collection.mutable
+import coop.rchain.roscala.util.Slot
+import coop.rchain.roscala.util.syntax._
 
 class TblObject extends Actor {
-  val keyVec = mutable.ArrayBuffer[Ob]()
+  val keyVec = Slot()
+
+  def entry(n: Int): Ob = extension.slot(n).get
+
+  def entryKey(n: Int): Ob = keyVec(n).get
 
   def addSlot(key: Ob, value: Ob): Int = {
-    keyVec.append(key)
-    super.addSlot(value)
+    val lock = keyVec.lock
+
+    //necessary because we want to update `keyVec` and `extension.slot` atomically
+    //otherwise another thread could try to access an uninitialized associated value
+    lock.writeLock().withLock {
```

Comment:
actually this is not atomic since keyVec and Actor.slot are two objects, and they are guard  by two separate lock.